Netable/Example/Services/AuthNetworkService.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  AuthNetworkService.swift
+//  Example
+//
+//  Created by Amy Oulton on 2022-11-28.
+//  Copyright © 2022 Steamclock Software. All rights reserved.
+//
+
+import Combine
+import Foundation
+import Netable
+
+class AuthNetworkService {
+    static var shared = AuthNetworkService()
+
+    private let unauthNetable: Netable
+
+    private var authNetable: Netable?
+    var netable: Netable {
+        authNetable ?? unauthNetable
+    }
+
+    var user: CurrentValueSubject<User?, Never>
+    var authError: CurrentValueSubject<NetableError?, Never>
+    var cancellables = [AnyCancellable]()
+
+    init() {
+        user = CurrentValueSubject<User?, Never>(nil)
+        authError = CurrentValueSubject<NetableError?, Never>(nil)
+
+        unauthNetable = Netable(
+            baseURL: URL(string: "http://localhost:8080/")!)
+
+        unauthNetable.requestFailurePublisher.sink { [weak self] error in
+            guard case let NetableError.httpError(statusCode, _) = error, statusCode == 401 else {
```

Comment:
nitpicky, but you can use a filter here instead of returning early to clean it up a little

Reply:
It's an enum and not an array, so instead of using `filter` I just did a simple `if` statement that if it doesn't pass nothing gets sent. If there is a better way to handle this I'd love to know how to improve it!